source/grid/tria.cc

```diff
@@ -950,8 +950,6 @@ namespace
     Assert(face_1->at_boundary() && face_2->at_boundary(),
            ExcMessage("Periodic faces must be on the boundary"));
 
-    Assert(std::abs(cell_1->level() - cell_2->level()) < 2, ExcInternalError());
```

Comment:
Shouldn't we leave this assertion in and instead check
```
Assert(dim == 1 || std::abs(cell_1->level() - cell_2->level()) < 2, ExcInternalError());
```

Reply:
@kronbichler I didn't remove it; I only moved it down into the else-statement. But I can keep it here; then it better fits to the @bangerth's explanation, which I will paraphrase.